stratify/_vinterp.pyx

```diff
@@ -557,7 +552,27 @@ cdef class _Interpolation(object):
 
         # Ensure that fz_src's shape is a superset of z_src's.
         if z_src.shape != fz_src.shape[-z_src.ndim:]:
-            raise ValueError('Shapes not consistent.')
+            emsg = 'Shape for z_src {} is not a subset of fz_src {}.'
+            raise ValueError(emsg.format(z_src.shape, fz_src.shape))
+
+        if z_target.ndim == 1:
+            z_target_size = z_target.shape[0]
+        else:
+            # Ensure z_target and z_src have same ndims.
+            if z_target.ndim != z_src.ndim:
+                emsg = ('z_target and z_src must have the same number '
+                        'of dimensions, got {} != {}.')
+                raise ValueError(emsg.format(z_target.ndim, z_src.ndim))
+            # Ensure z_target and z_src have same shape over their
+            # non-interpolated axes.
+            zts, zss = z_target.shape, z_src.shape
+            ztsp, zssp = zip(*[(str(j), str(k)) if i!=zp_axis else (':', ':')
```

Comment:
Quite subtle, I didn't realise the context of why you had done this until I saw the exception message from one of the unittests, perhaps a comment with context (example exception message) would be useful here??